src/test/java/com/cst438/controller/SectionControllerSystemTest.java

```diff
@@ -232,36 +232,36 @@ public void systemTestAddSectionBadCourse() throws Exception {
         // clear the courseId field and enter cst499
         WebElement courseId = driver.findElement(By.id("ecourseId"));
         courseId.sendKeys(Keys.chord(Keys.CONTROL,"a", Keys.DELETE));
-       Thread.sleep(SLEEP_DURATION);
+        Thread.sleep(SLEEP_DURATION);
         courseId.sendKeys("cst499");
         driver.findElement(By.id("save")).click();
         Thread.sleep(SLEEP_DURATION);
 
-       message = driver.findElement(By.id("addMessage")).getText();
-       assertTrue(message.startsWith("section added"));
+        message = driver.findElement(By.id("addMessage")).getText();
+        assertTrue(message.startsWith("section added"));
 
-       // close the dialog
-       driver.findElement(By.id("close")).click();
-       Thread.sleep(SLEEP_DURATION);
+        // close the dialog
+        driver.findElement(By.id("close")).click();
+        Thread.sleep(SLEEP_DURATION);
 
         WebElement row = driver.findElement(By.xpath("//tr[td='cst499']"));
         assertNotNull(row);
-       // find the delete button on the row from prior statement.
-       List<WebElement> deleteButtons = row.findElements(By.tagName("button"));
-       // delete is the second button
-       assertEquals(2, deleteButtons.size());
-       deleteButtons.get(1).click();
-       Thread.sleep(SLEEP_DURATION);
-       // find the YES to confirm button
-       List<WebElement> confirmButtons = driver
-               .findElement(By.className("react-confirm-alert-button-group"))
-               .findElements(By.tagName("button"));
-       assertEquals(2,confirmButtons.size());
-       confirmButtons.get(0).click();
-       Thread.sleep(SLEEP_DURATION);
-
-       // verify that Section list is empty
-       assertThrows(NoSuchElementException.class, () ->
-               driver.findElement(By.xpath("//tr[td='cst499']")));
+        // find the delete button on the row from prior statement.
+        List<WebElement> deleteButtons = row.findElements(By.tagName("button"));
+        // delete is the second button
+        assertEquals(2, deleteButtons.size());
+        deleteButtons.get(1).click();
+        Thread.sleep(SLEEP_DURATION);
+        // find the YES to confirm button
+        List<WebElement> confirmButtons = driver
+                .findElement(By.className("react-confirm-alert-button-group"))
+                .findElements(By.tagName("button"));
+        assertEquals(2,confirmButtons.size());
+        confirmButtons.get(0).click();
+        Thread.sleep(SLEEP_DURATION);
+
+        // verify that Section list is empty
+        assertThrows(NoSuchElementException.class, () ->
+                driver.findElement(By.xpath("//tr[td='cst499']")));
```

Comment:
Catching with NoSuchElementException and continuing with the test is fine but if the element is supposed to be there, the test should fail when the exception is thrown.